src/webgpu/api/operation/memory_sync/buffer/buffer_sync_test.ts

```diff
@@ -114,14 +329,34 @@ export class BufferSyncTest extends GPUTest {
     });
   }
 
+  // Create a render pipeline and write given data into storage buffer at fragment stage.
+  createStorageWriteRenderPipeline(value: number): GPURenderPipeline {
+    const wgslShaders = {
+      vertex: kDummyVertexShader,
+      fragment: `
+      struct Data {
+        a : u32;
+      };
+
+      @group(0) @binding(0) var<storage, read_write> data : Data;
+      @stage(fragment) fn frag_main() -> @location(0) vec4<f32> {
+        data.a = ${value}u;
+        return vec4<f32>(1.0, 0.0, 0.0, 1.0);
```

Comment:
does it matter that it's red or can we just return vec4<f32>(); // result doesn't matter
?